README.md

```diff
@@ -47,9 +47,14 @@ variations.
 
 Each test case has the the following keys:
 - description: which will be used to name each generated test
+  - the description should not simply explain **what** each case is (that is redundant information)
+  - the description should explain **why** each case is there
+  - for example, what kinds of incorrect implementations might this case help us find?
 - 'variable names': one or more variable names with values which will be passed to the solution method
-- expected: the expected result (this would be -1 when we expect an exception)
-- msg: a nice message for the failing case
+- expected: the expected result
+  - if an error is expected, the value at `"expected"` should be an object containing exactly one property, `"error"`, whose value is a string.
```

Comment:
In this proposal, how should one differentiate between a missing value (optional return) and an error? Is it possible, or both should use the same structure?